src/EnergyPlus/Material.cc

```diff
@@ -257,7 +257,10 @@ void GetMaterialData(EnergyPlusData &state, bool &ErrorsFound) // set to true if
             MaterNum = ip->getIDFObjNum(state, state.dataHeatBalMgr->CurrentModuleObject, counter);
 
             // Load the material derived type from the input data.
-            auto thisMaterial = state.dataMaterial->Material(MaterNum);
+            delete state.dataMaterial->Material(MaterNum);
+            state.dataMaterial->Material(MaterNum) = new Material::MaterialChild;
+            auto *thisMaterial = dynamic_cast<Material::MaterialChild *>(state.dataMaterial->Material(MaterNum));
```

Comment:
Don't need to downcast if you assign `thisMaterial` to `new Material::MaterialChild first`.

Reply:
right right. That's better